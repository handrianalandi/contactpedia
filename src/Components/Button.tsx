import React from 'react'
import { Button as ButtonElement } from 'react-bootstrap'
import Spinner from './Spinner';

interface ButtonProps {
    children?: React.ReactNode;
    onClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
    variant?: string;
    isLoading?: boolean;
}
export default function Button({isLoading=false,children, ...restProps}: ButtonProps) {
  return (
    <ButtonElement 
        {...restProps}
        disabled={isLoading}
    >
        {isLoading ? <Spinner size='sm' isInvertedColor/> : children}
    </ButtonElement>
  )
}
