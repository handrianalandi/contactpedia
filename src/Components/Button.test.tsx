import {render,screen} from '@testing-library/react';
import Button from './Button';
describe('Button', () => {
    //create jest fn
    it('should render correctly', () => {
        const {container} = render(<Button />);
        expect(container).toMatchSnapshot();
    });

    it('should render correctly with children', () => {
        const {container} = render(<Button>Test</Button>);
        expect(container).toMatchSnapshot();
    }
    );

    it('should render correctly with isLoading', () => {
        const {container} = render(<Button isLoading/>);
        expect(container).toMatchSnapshot();
    }
    );

    it('should render correctly with variant', () => {
        const {container} = render(<Button variant='success'/>);
        expect(container).toMatchSnapshot();
    }
    );

    it('should render correctly with onClick', () => {
        const mockOnClick = jest.fn();
        render(<Button onClick={mockOnClick}/>);
        //click the button
        screen.getByRole('button').click();
        expect(mockOnClick).toHaveBeenCalled();
    }
    );
});