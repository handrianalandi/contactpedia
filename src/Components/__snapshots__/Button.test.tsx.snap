// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<div>
  <button
    class="btn btn-primary"
    type="button"
  />
</div>
`;

exports[`Button should render correctly with children 1`] = `
<div>
  <button
    class="btn btn-primary"
    type="button"
  >
    Test
  </button>
</div>
`;

exports[`Button should render correctly with isLoading 1`] = `
<div>
  <button
    class="btn btn-primary"
    disabled=""
    type="button"
  >
    <div>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-spinner fa-spin fa-sm  css-1vbona3"
        data-icon="spinner"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M304 48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zm0 416a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM48 304a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm464-48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM142.9 437A48 48 0 1 0 75 369.1 48 48 0 1 0 142.9 437zm0-294.2A48 48 0 1 0 75 75a48 48 0 1 0 67.9 67.9zM369.1 437A48 48 0 1 0 437 369.1 48 48 0 1 0 369.1 437z"
          fill="currentColor"
        />
      </svg>
    </div>
  </button>
</div>
`;

exports[`Button should render correctly with variant 1`] = `
<div>
  <button
    class="btn btn-success"
    type="button"
  />
</div>
`;
